### PR TITLE
refactor: reorganise contracts into separate files

### PR DESCRIPTION
**Pull request by superical**
## Summary
Reorganise the contracts into their separate files instead of having everything in a single huge file for readability and maintainability. Also to prep a little so that it's easier later upgrading to OZ4.

## Changes
* Separate out all contracts and interfaces into their own files
* Add folders to organise the contracts
* Move all flattened dependencies into a `lib` folder

### CODE REVIEW

File: contracts/TitleEscrowCloneable.sol [hunk 1]
```diff
@@ -1,58 +1,18 @@
...
-    return _msgSender() == holder;
-  }
-
-  function _changeHolder(address newHolder) internal {
-    require(newHolder != address(0), "HasHolder: new holder is the zero address");
-    emit HolderChanged(holder, newHolder);
-    holder = newHolder;
-  }
-}
+import "./access/HasNamedBeneficiaryInitializable.sol";
```
Comment (Nebulis) [inline contracts/TitleEscrowCloneable.sol:4]: any consequences on the contract size (and deployment cost) ?

**Review by Nebulis — APPROVED** [review]

File: contracts/TitleEscrowCloneable.sol (hunk 1 above)
Comment (superical) [reply contracts/TitleEscrowCloneable.sol:4]: Did you mean the whether separating the files will increase the contract bytecode size? I think it shouldn't. Just to be safe, I did a quick check on their sizes between before and after this change and they are the same. Here are the numbers for fun:
```
Size in KB.
·························|··············
 |  TitleEscrowCloneable  ·      5.521  │
 ·························|··············
 |  TitleEscrowCloner     ·      0.669  │
 ·························|··············
 |  TradeTrustERC721      ·     10.031  │
 ·------------------------|-------------·
```

Comment (superical) [reply contracts/TitleEscrowCloneable.sol:4]: > Also we will need to republish the contract in etherscan =)

Ohh!😱 We have contracts deployed on mainnet?

Comment (Nebulis) [reply contracts/TitleEscrowCloneable.sol:4]: that's what happens when you split files and interfaces haha :D

Do you think automation could be part of the build on master? post release, only if there is a contract change?

Comment (Nebulis) [reply contracts/TitleEscrowCloneable.sol:4]: Hum not sure you understood me correctly
I think we can leave the library release cycle as is

I was referring to the smart contract releases. I think the ideal solution would be to be able to deploy on Etherscan the contract source when there is any change on it.

**Pull request merged**